packages/gatsby-cli/src/reporter/index.js

```diff
@@ -343,6 +343,25 @@ const reporter: Reporter = {
     }
     const span = tracer.startSpan(text, spanArgs)
 
+    let lastUpdateTime = 0
+    let progress = 0
+    const progressUpdateDelay = Math.round(1000 / 10) // 10 fps *shrug*
+
+    const updateProgress = () => {
+      const t = Date.now()
+      if (t - lastUpdateTime <= progressUpdateDelay) return
+
+      if (total > 0) {
```

Comment:
if this is the way you want to track if something need to be dispatched, we do need to set `total` to 0 in `progressActivity.start()` after calling `reporterActions.startActivity`, as this will cause emitting unnecessary `total` event

Reply:
Ok good point, I'll add another var for it.